cheatsheets/Forgot_Password_Cheat_Sheet.md

```diff
@@ -1,60 +1,104 @@
-# Introduction
+# Forgot Password and MFA Cheat Sheet
 
-This article provides a simple model to follow when implementing a *forgot password* web application feature.
+## Introduction
 
-# The Problem
+In order to implement a proper user management system, almost all systems integrate a **Forgot Password/MFA?** service, which allows the user to request a reset to their password or MFA whenever they forget any of them, or if their account ever gets breached.
 
-There is no industry standard for implementing a **Forgot Password** feature. The result is that you see applications forcing users to jump through myriad hoops involving emails, special URLs, temporary passwords, personal security questions, and so on. In the end you have to reset it to a new value.
+As much as is this functionality looks straight-forward and easy to implement, the details of its implementation makes it a sweet spot for security attacks, such as the renowned [user enumeration attack](https://owasp.org/www-project-web-security-testing-guide/stable/4-Web_Application_Security_Testing/03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account.html).
 
-# Steps
+The following short guidelines can be used as a quick reference to protect the forgot password or MFA service:
 
-## Step 1) Gather Identity Data or Security Questions
+- **Unified user message, whether the user exists or not.**
+- **Ensure that the time taken for the user response message is uniform.**
```

Comment:
I find it weird that these 2 items are not mentioned in the rest of the document but the third item is. This part reads like it's a TLDR of the whole document but that's not how it actually works currently. IMO this "quick reference" should be essentially a TLDR of the main ideas from the rest of the document. These 2 maybe belong under the Operational Tasks section at the bottom.

Being a person with appsec experience I understand exactly what these mean but a developer or new appsec person looking at their first or second cheat sheet might not. What's something short that can be added to help with that? Maybe a link to or mention of user enumeration for item one and timing attacks for item two.